smartsim/_core/mli/infrastructure/environmentloader.py

```diff
@@ -41,27 +40,50 @@ class EnvironmentConfigLoader:
     into the WorkerManager.
     """
 
-    def __init__(self) -> None:
-        self._feature_store_descriptor: t.Optional[str] = os.getenv(
-            "SSFeatureStore", None
-        )
+    def __init__(
+        self,
+        featurestore_factory: t.Callable[[str], FeatureStore],
+        callback_factory: t.Callable[[bytes], CommChannelBase],
+        queue_factory: t.Callable[[str], CommChannelBase],
+    ) -> None:
         self._queue_descriptor: t.Optional[str] = os.getenv("SSQueue", None)
-        self.feature_store: t.Optional[FeatureStore] = None
-        self.queue: t.Optional[DragonFLIChannel] = None
+        """The descriptor used to attach to the incoming event queue"""
+        self.queue: t.Optional[CommChannelBase] = None
+        """The attached incoming event queue channel"""
+        self._backbone_descriptor: t.Optional[str] = os.getenv("SS_DRG_DDICT", None)
+        """The descriptor used to attach to the backbone feature store"""
+        self.backbone: t.Optional[FeatureStore] = None
+        """The attached backbone feature store"""
+        self._featurestore_factory = featurestore_factory
+        """A factory method to instantiate a FeatureStore"""
+        self._callback_factory = callback_factory
+        """A factory method to instantiate a concrete CommChannelBase
+        for inference callbacks"""
+        self._queue_factory = queue_factory
+        """A factory method to instantiate a concrete CommChannelBase
+        for inference requests"""
 
-    def get_feature_store(self) -> t.Optional[FeatureStore]:
-        """Loads the Feature Store previously set in SSFeatureStore"""
-        if self._feature_store_descriptor is not None:
-            self.feature_store = pickle.loads(
-                base64.b64decode(self._feature_store_descriptor)
-            )
-        return self.feature_store
+    def get_backbone(self) -> t.Optional[FeatureStore]:
+        """Create the backbone feature store using the descriptor found in
+        an environment variable"""
```

Comment:
appended description from earlier item into the property docstring. i think this might satisfy this request:
```
        """Create the backbone feature store using the descriptor found in
        an environment variable. The backbone is a standalone, system-created
        feature store used to share internal information among MLI components"""
```